Tests/RevenueCatUITests/CustomerCenter/ManageSubscriptionsViewModelTests.swift

```diff
@@ -71,27 +73,256 @@ class ManageSubscriptionsViewModelTests: TestCase {
         expect(viewModel.isLoaded) == true
     }
 
-    func testLoadScreenSuccess() async {
+    func testShouldShowActiveSubscription_whenUserHasOneActiveSubscriptionOneEntitlement() async {
+        // Arrange
+        let productId = "com.revenuecat.product"
+        let purchaseDate = "2022-04-12T00:03:28Z"
+        let expirationDate = "2062-04-12T00:03:35Z"
+        let products = [Fixtures.product(id: productId, title: "title", duration: .month, price: 2.99)]
+        let customerInfo = Fixtures.customerInfo(
+            subscriptions: [
+                Fixtures.Subscription(
+                    id: productId,
+                    store: "app_store",
+                    purchaseDate: purchaseDate,
+                    expirationDate: expirationDate
+                )
+            ],
+            entitlements: [
+                Fixtures.Entitlement(
+                    entitlementId: "premium",
+                    productId: productId,
+                    purchaseDate: purchaseDate,
+                    expirationDate: expirationDate
+                )
+            ]
+        )
+
         let viewModel = ManageSubscriptionsViewModel(screen: ManageSubscriptionsViewModelTests.screen,
-                                                     purchasesProvider: MockManageSubscriptionsPurchases())
+                                                     purchasesProvider: MockManageSubscriptionsPurchases(
+                                                        customerInfo: customerInfo,
+                                                        products: products
+                                                     ))
 
+        // Act
         await viewModel.loadScreen()
 
+        // Assert
         expect(viewModel.subscriptionInformation).toNot(beNil())
         expect(viewModel.screen).toNot(beNil())
         expect(viewModel.state) == .success
 
         expect(viewModel.subscriptionInformation?.title) == "title"
         expect(viewModel.subscriptionInformation?.durationTitle) == "month"
         expect(viewModel.subscriptionInformation?.price) == "$2.99"
-        expect(viewModel.subscriptionInformation?.nextRenewalString) == "Apr 12, 2062"
-        expect(viewModel.subscriptionInformation?.productIdentifier) == "com.revenuecat.product"
+
+        let formatter = DateFormatter()
+        formatter.dateStyle = .medium
+        expect(viewModel.subscriptionInformation?.nextRenewalString) ==
+            formatter.string(from: ISO8601DateFormatter().date(from: expirationDate)!)
```

Comment:
we're doing `nextRenewalString` == string from date from string, maybe there's a shortcut we can use here? How are we formatting that renewalString in the first place?

Reply:
Yes we are basically re-formatting a ISO-8601 string representation of a date. I'm not sure if there's a shortcut here, but it can definitely be extracted to a separate method, which is what I did here: https://github.com/RevenueCat/purchases-ios/pull/4046/commits/41b8cc054928544833b8f7aaa0cb1c155d3e3136

The formatting of the `nextRenewalString` is unchanged and is as follows:
```swift
let dateFormatter = DateFormatter()
dateFormatter.dateStyle = .medium
let nextRenewalString = currentEntitlement.expirationDate.map { dateFormatter.string(from: $0) }
```